### ISSUE TITLE
Onnx export with python models/export.py --dynamic does not produce dynamic outputs

### ISSUE THREAD

**Issue by SamSamhuns** [ultralytics/yolov5#3444]
## 🐛 Bug
Exporting the pre-trained `yolov5s` model with the `--dynamic` option results in an onnx model with dynamic inputs however the outputs are not dynamic. This creates an issue when using the onnx model in another ml network server such as triton-server later on.

The `torch.onnx.export` call in `models/export.py` does not set the output_names and the `output` parameter for the dynamic axes has no affect on the final output nodes.

## To Reproduce

```
git clone https://github.com/ultralytics/yolov5                              # clone repo
cd yolov5
pip install -r requirements.txt                                              # base requirements
pip install coremltools>=4.1 onnx>=1.9.0 scikit-learn==0.19.2                # export requirements
python models/export.py --dynamic                                            # export yolov5s model as onnx
python -c "import onnx; m=onnx.load('./yolov5s.onnx'); print(m.graph.input); print(m.graph.output)". # print input and output shape of onnx model
```

Output:
```
[name: "images"
type {
  tensor_type {
    elem_type: 1
    shape {
      dim {dim_param: "batch"}
      dim {dim_value: 3}
      dim {dim_param: "height"}
      dim {dim_param: "width"}
   }
  }
}
]
[name: "672"
type {
  tensor_type {
    elem_type: 1
    shape {
      dim {dim_value: 1}
      dim {dim_value: 25200}
      dim {dim_value: 85}
    }
  }
}
, name: "397"
type {
  tensor_type {
    elem_type: 1
    shape {
      dim {dim_value: 1}
      dim {dim_value: 3}
      dim {dim_value: 80}
      dim {dim_value: 80}
      dim {dim_value: 85}
    }
  }
}
, name: "495"
type {
  tensor_type {
    elem_type: 1
    shape {
      dim {dim_value: 1}
      dim {dim_value: 3}
      dim {dim_value: 40}
      dim { dim_value: 40}
      dim {dim_value: 85}
    }
  }
}
, name: "593"
type {
  tensor_type {
    elem_type: 1
    shape {
      dim {dim_value: 1}
      dim {dim_value: 3}
      dim {dim_value: 20}
      dim {dim_value: 20}
      dim {dim_value: 85}
    }
  }
}
]
```

## Expected behavior

The output nodes should have dynamic output axes called `batch` instead of having the first output shape as `dim {dim_value: 1}`
i.e.
```
[name: "672"
type {
  tensor_type {
    elem_type: 1
    shape {
      dim {dim_param: "batch"}
      dim {dim_value: 25200}
      dim {dim_value: 85}
    }
  }
}
..... and so on for other output nodes
```

## Environment

 - OS: Ubuntu 18.04.5
 - GPU Tesla V100

## My current solution

Add `output_names` parameter to torch.onnx export and set the correct values for the outputs in the `dynamic_axes`

```
torch.onnx.export(model, img, f, verbose=False, opset_version=opt.opset_version, input_names=['images'],
                              training=torch.onnx.TrainingMode.TRAINING if opt.train else torch.onnx.TrainingMode.EVAL,
                              do_constant_folding=not opt.train,
                              output_names=['output0', 'output1', 'output2', 'output3'],
                              dynamic_axes={'images': {0: 'batch', 2: 'height', 3: 'width'},  # size(1,3,640,640)
                                            'output0': {0: 'batch'}, 'output1': {0: 'batch'},
                                            'output2': {0: 'batch'}, 'output3': {0: 'batch'}} if opt.dynamic else None)
```

**Comment by glenn-jocher** [comment]
@SamSamhuns thanks for the bug report! The fastest way to get your fixes into the code is to submit a PR directly. Once you do this we can review and comment there. Thank you!

**Comment by SamSamhuns** [comment]
Hi @glenn-jocher, I have created a PR #3456 . Please have a look, thanks.
